app/models/namespaces/project_namespace.rb

```diff
@@ -34,5 +34,19 @@ def of_ancestors_and_self
     def self.sti_name
       'Project'
     end
+
+    def self_and_parents
+      namespaces = [self]
+      namespaces += parent.self_and_ancestors unless parent.type == 'User'
+      namespaces
+    end
+
+    def update_metadata_summary_by_update_service(metadata_to_subtract, metadata_to_add)
+      namespaces_to_update = self_and_parents
```

Comment:
I don't think we need a separate method for self_and_parents. We already have a query to get the ancestors. You should be able to just call `namespaces_to_update = [self] + parent.self_and_ancestors`

Reply:
Thanks for this, much better. Changed in 398dc863c9111dce05cac4a08fa3f29ea205ad0e